crud/src/main/java/com/redhat/lightblue/rest/crud/hystrix/GenerateCommand.java

```diff
@@ -0,0 +1,131 @@
+    private static final Logger LOGGER = LoggerFactory.getLogger(GenerateCommand.class);
+
+    private final String entity;
+    private final String version;
+    private final String field;
+    private final int n;
+
+    public GenerateCommand(String clientKey, String entity, String version, String field,int n) {
+        super(GenerateCommand.class, clientKey,null);
+        this.entity = entity;
+        this.version = version;
+        this.field = field;
+        this.n = n<=0?1:n;
+    }
+    
+    @Override
+    protected CallStatus run() {
+        LOGGER.debug("run: entity={}, version={}", entity, version);
+        Error.reset();
+        Error.push("rest");
+        Error.push(getClass().getSimpleName());
+        Error.push(entity);
+        try {
+            LightblueFactory lbf=RestConfiguration.getFactory();
+            Factory factory=lbf.getFactory();
+            Metadata md=lbf.getMetadata();
+            DefaultMetadataResolver mdResolver=new DefaultMetadataResolver(md);
+            Path fieldPath=new Path(field);
+            mdResolver.initialize(entity,version,null,new FieldProjection(fieldPath,true,false));
+            HttpServletRequest sreq=ResteasyProviderFactory.getContextData(HttpServletRequest.class);
+            Set<String> userRoles=new HashSet<>();
+            if(sreq!=null) {
+                for(String role:mdResolver.getMetadataRoles()) {
+                    if(sreq.isUserInRole(role))
+                        userRoles.add(role);
+                }
+            }
+            LOGGER.debug("user roles:{}",userRoles);
+            EntityMetadata emd=mdResolver.getCompositeMetadata();
+            if(emd.getAccess().getFind().hasAccess(userRoles)||
+               emd.getAccess().getInsert().hasAccess(userRoles)||
+               emd.getAccess().getUpdate().hasAccess(userRoles)) {
+                LOGGER.debug("User has access, looking up {}",fieldPath);
+                FieldTreeNode ftn=emd.resolve(fieldPath);
+                LOGGER.debug("Field={}",ftn);
+                SimpleField fld=(SimpleField)ftn;
+                LOGGER.debug("Generator:{}",fld.getValueGenerator());
+                if(fld.getValueGenerator()!=null) {
+                    LOGGER.debug("{} has generator: {}",fieldPath,fld.getValueGenerator().getValueGeneratorType());
+                    ArrayNode arr=JsonNodeFactory.instance.arrayNode();
+                    for(int i=0;i<n;i++) {
+                        JsonNode node=GeneratedFields.generate(factory,fld,emd);
+                        arr.add(node);
+                    }
+                    com.redhat.lightblue.Response r=new com.redhat.lightblue.Response();
+                    r.setStatus(OperationStatus.COMPLETE);
+                    r.setEntityData(arr);
+                    return new CallStatus(r);
+                }
+            } else {
+                throw Error.get(CrudConstants.ERR_NO_ACCESS, "insert " + mdResolver.getTopLevelEntityName());
```

Comment:
Copy/paste error. Fixed.

On Wed, May 4, 2016 at 12:44 PM, Naveen Malik <notifications@github.com>
wrote:

> In
> crud/src/main/java/com/redhat/lightblue/rest/crud/hystrix/GenerateCommand.java
> <https://github.com/lightblue-platform/lightblue-rest/pull/229#discussion_r62092674>
> :
>
> > +                SimpleField fld=(SimpleField)ftn;
> > +                LOGGER.debug("Generator:{}",fld.getValueGenerator());
> > +                if(fld.getValueGenerator()!=null) {
> > +                    LOGGER.debug("{} has generator: {}",fieldPath,fld.getValueGenerator().getValueGeneratorType());
> > +                    ArrayNode arr=JsonNodeFactory.instance.arrayNode();
> > +                    for(int i=0;i<n;i++) {
> > +                        JsonNode node=GeneratedFields.generate(factory,fld,emd);
> > +                        arr.add(node);
> > +                    }
> > +                    com.redhat.lightblue.Response r=new com.redhat.lightblue.Response();
> > +                    r.setStatus(OperationStatus.COMPLETE);
> > +                    r.setEntityData(arr);
> > +                    return new CallStatus(r);
> > +                }
> > +            } else {
> > +                throw Error.get(CrudConstants.ERR_NO_ACCESS, "insert " + mdResolver.getTopLevelEntityName());
>
> /insert/generate/
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-rest/pull/229/files/0bc69428bb58e29e04c0745b8fdbf722b633a7bc#r62092674>
>